### PR TITLE
 Provide a way to perform java script code in a readers page from external code

### PR DESCRIPTION
**Pull request by tschob**
This PR:
- refactors the `UIWebView` extension to the subclass `FolioReaderWebView`
- Provides a way to let extneral code run java script in a `FolioReaderPage`

### CODE REVIEW

File: Source/FolioReaderPage.swift [hunk 1]
```diff
@@ -157,7 +157,7 @@ public class FolioReaderPage: UICollectionViewCell, UIWebViewDelegate, UIGesture
         }
         
         UIView.animateWithDuration(0.2, animations: {webView.alpha = 1}) { finished in
-            webView.isColors = false
+            (webView as? FolioReaderWebView)?.isColors = false
```
Comment (hebertialmeida) [inline Source/FolioReaderPage.swift]: Lets reduce the `(webView as? FolioReaderWebView)` cast, we can just cast with guard on beginning of the function.

``` swift
guard let webView = webView as? FolioReaderWebView else { return }
```

Then just use `webView` as previously.

File: Source/FolioReaderPage.swift [hunk 2]
```diff
@@ -175,8 +175,8 @@ public class FolioReaderPage: UICollectionViewCell, UIWebViewDelegate, UIGesture
             let decoded = url.absoluteString.stringByRemovingPercentEncoding as String!
             let rect = CGRectFromString(decoded.substringFromIndex(decoded.startIndex.advancedBy(12)))
             
-            webView.createMenu(options: true)
-            webView.setMenuVisible(true, andRect: rect)
+            (webView as? FolioReaderWebView)?.createMenu(options: true)
+            (webView as? FolioReaderWebView)?.setMenuVisible(true, andRect: rect)
```
Comment (hebertialmeida) [inline Source/FolioReaderPage.swift]: Same for casting here and so on...

**Review by hebertialmeida — CHANGES_REQUESTED** [review]
Why do you think is better to subclass the `UIWebView` instead of a extension, as it is not public to override?

**Comment by tschob** [comment]
I think it's better to use a subclass instead of an extension because:
- The added logic is only needed in one place and it only contains logic which is needed for the reader and useless for other use cases. With this needs it should normally be a subclass and not an extension.
- The extension effects all `UIWebView` instances in an app which uses the Folio Reader. If Folio Readers `UIMenuController` isn't overridden it's possible to select text in a app owned `UIWebView` instance and choose eg. "Highlight" (as the `UIWebView` extension says it can perform this action).
- Folio Readers `UIWebView` is less customisable as extension as everything which is added will effect all `UIWebView` instances in an app. So only things which don't break other `UIWebView` instances can be added. The old implementation breaks other instances.

File: Source/FolioReaderPage.swift (hunk 1 above)
Comment (tschob) [reply Source/FolioReaderPage.swift]: Yes this seems to be the better approach, I'll adjust it.

**Review by hebertialmeida — APPROVED** [review]

**Pull request merged**